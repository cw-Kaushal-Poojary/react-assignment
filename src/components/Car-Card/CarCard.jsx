import styles from "./car-cards.module.css";

// ASSETS //
import testCar from "../../assets/test-car.jpg";

const CarCard = ({ car }) => {
  return (
    <div className={styles["car-card"]}>
      <div className={styles["card-inner"]}>
        <img
          className={styles["card-image"]}
          src={car?.imageUrl || testCar}
          alt="car"
          height={100}
          width={100}
        />
        <div className={styles["card-content"]}>
          <h3 className={styles["card-title"]}>{car?.carName}</h3>
          <p className={styles["card-description"]}>
            {car?.kmNumeric}km | {car.fuel} | {car?.areaName}, {car?.cityName}
          </p>
          <p className={styles["price"]}>Rs {car?.price}</p>
          <div className={styles["btn-wrap"]}>
            <button className={styles["seller-btn"]}>Get Seller Details</button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CarCard;
