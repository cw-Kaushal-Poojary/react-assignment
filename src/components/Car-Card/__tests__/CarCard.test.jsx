import React from 'react';
import { render, screen } from '@testing-library/react';
import CarCard from '../CarCard';
import '@testing-library/jest-dom';

describe('CarCard Component', () => {
  const car = {
    imageUrl: "test-image.jpg",
    carName: "Test Car",
    kmNumeric: "20000",
    fuel: "Petrol",
    areaName: "Test Area",
    cityName: "Test City",
    price: "5 Lakh"
  };

  test('renders car details correctly', () => {
    render(<CarCard car={car} />);

    expect(screen.getByText('Test Car')).toBeInTheDocument();
    expect(screen.getByText('20000km | Petrol | Test Area, Test City')).toBeInTheDocument();
    expect(screen.getByText('Rs 5 Lakh')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<CarCard car={car} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
