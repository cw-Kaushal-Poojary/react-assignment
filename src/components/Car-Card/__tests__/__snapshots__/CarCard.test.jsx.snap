// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <img
        alt="car"
        height="100"
        src="test-image.jpg"
        width="100"
      />
      <div>
        <h3>
          Test Car
        </h3>
        <p>
          20000km | Petrol | Test Area, Test City
        </p>
        <p>
          Rs 5 Lakh
        </p>
        <div>
          <button>
            Get Seller Details
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
